aten/src/TH/generic/THTensorEvenMoreMath.cpp

```diff
@@ -138,20 +138,35 @@ void THTensor_(nonzero)(THLongTensor *subscript, THTensor *tensor)
 #endif
   THLongTensor_resize2d(subscript, numel, tensor->dim());
 
+  int64_t dimensions = tensor->dim();
+  int64_t sizes[dimensions];
+  int64_t idx[dimensions];
+  for (i = 0; i < dimensions; ++i) {
+    sizes[i] = THTensor_sizeLegacyNoScalars(tensor, i);
```

Comment:
This won't break, but doesn't make a whole lot of sense.  THTensor_sizeLegacyNoScalars is for getting 1 instead of 0 if you have a 0-dimensional tensor, but you can only enter this loop if it's not a 0-dimensional tensor.

Reply:
Resolved by adding return statement.